### PR TITLE
#333 properties table performance

### LINKED ISSUE #333: Common Properties unable to render extremely large values
There is a use case where we're trying to render a list of values in a textarea control but that was crashing the browser. One of the values in the array is more than 50 million characters long. 
image
image

We need to limit the length of strings getting displayed in the UI. If the value is too long, it should be displayed in a readonly mode where it cannot be edited. This applies to textarea controls standalone and in structures, as well as lists

### PR DESCRIPTION
**Pull request by caritaou**
fixes #333
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/__tests__/common-properties/controls/structuretable-test.js [hunk 1]
```diff
@@ -1029,3 +1029,34 @@ describe("measurement icons should be rendered correctly in structuretable", ()
 		expect(tableWrapper.find("div.properties-field-type-icon")).to.have.length(0);
 	});
 });
+
+describe("structuretable with long text input values should render as readonly", () => {
```
[the remark this answers is not in the document]
Comment (matthoward366) [reply canvas_modules/common-canvas/__tests__/common-properties/controls/structuretable-test.js:1033]: It's kind of hard to find in the structuretable paramDef.

[the remark this answers is not in the document]
Comment (caritaou) [reply canvas_modules/common-canvas/__tests__/common-properties/controls/structuretable-test.js:1033]: This was to test that the error appears correctly within a table cell and in subpanel of a structure. I should add a test in textfield and textarea as well

[the remark this answers is not in the document]
Comment (matthoward366) [reply canvas_modules/common-canvas/__tests__/common-properties/controls/structuretable-test.js:1033]: I think that makes sense.  At least have a paramDef for each.

**Review by matthoward366 — APPROVED** [review]
Looks good.

**Pull request merged**